subprojects/docs/src/samples/templates/standalone-precompiled-script-plugin-utils/src/main/java/com/example/ReadmeVerificationTask.java

```diff
@@ -0,0 +1,34 @@
+package com.example;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.util.GFileUtils;
+
+import java.util.regex.Pattern;
+
+public abstract class ReadmeVerificationTask extends DefaultTask {
+
+    @PathSensitive(PathSensitivity.RELATIVE)
+    @InputFile
+    public abstract RegularFileProperty getReadme();
+
+    @Internal
+    public abstract ListProperty<String> getReadmePatterns();
+
+    @TaskAction
+    void verifyServiceReadme() {
+        var readmeContents = GFileUtils.readFile(getReadme().getAsFile().get());
```

Comment:
:x: `GFileUtils` is internal api, better not to use it in a sample. Assuming we're using Java >= 7 it could be rewritten as follow:
```suggestion
        var readmeContents = new String(Files.readAllBytes(getReadme().getAsFile().get().toPath()), UTF_8)
```

Reply:
Since I'm already using Java 11 `var` here, I can also do `Files.readString(getReadme().getAsFile().get().toPath())`. Wasn't aware `GFileUtils` is internal API.